toolkit/offchain/src/cardano_keys.rs

```diff
@@ -46,22 +46,22 @@ impl CardanoPaymentSigningKey {
 	}
 }
 
-impl TryFrom<CardanoKeyFileContent> for CardanoPaymentSigningKey {
+impl TryFrom<CardanoKeyFileContent> for CardanoSigningKey {
 	type Error = anyhow::Error;
 
 	fn try_from(key: CardanoKeyFileContent) -> Result<Self, Self::Error> {
 		let key_type = key.r#type.clone();
-		if key_type == "PaymentSigningKeyShelley_ed25519" {
-			Ok(CardanoPaymentSigningKey::from_normal_bytes(key.raw_key_bytes()?)?)
+		if key_type == "PaymentSigningKeyShelley_ed25519" || key_type == "StakeSigningKeyShelley_ed25519" {
```

Comment:
Maybe I should have commented this, but I have not allowed StakeSigningKey here for a reason.
It was to prevent using cardano staking key when payment key is expected.

So, please leave `CardanoPaymentSigningKey` intact and you could implement `TryFrom<CardanoKeyFileContent>` for `MainchainPrivateKey` which is staking key.

Reply:
I think we just don't need this addition here and also nothing to do with MainchainPrivateKey.